src/data/data.hpp

```diff
@@ -85,33 +84,38 @@ struct TemperatureData : public Sensor {
 struct Sensors : public Module {
   static constexpr int kNumImus = 4;
   static constexpr int kNumKeyence = 2;
+  static constexpr int kFIFO;
+  
+  vector<NavigationVector> imu_vector(kFIFO);
 
   DataPoint<array<ImuData, kNumImus>> imu;
   array<StripeCounter, kNumKeyence>  keyence_stripe_counter;
 };
 
 struct BatteryData {
-  static constexpr int kNumCells = 36;
-  uint16_t  voltage;                    // dV
-  int16_t   current;                    // dA
-  uint8_t   charge;                     // %
-  int8_t    average_temperature;        // C
+  uint16_t             voltage;                    // dV
+  uint16_t             current;                    // dA
+  uint16_t             charge;                     // %
+  uint16_t             average_temperature;        // C
+};
 
+struct HPBatteryData : public BatteryData {
   // below only for BMSHP! Value for BMSLP = 0
-  uint16_t  cell_voltage[kNumCells];    // mV
-  int8_t    low_temperature;            // C
-  int8_t    high_temperature;           // C
-  uint16_t  low_voltage_cell;           // mV
-  uint16_t  high_voltage_cell;          // mV
-  bool      imd_fault;
+  static constexpr int kNumCells = 36;
+  uint16_t             hp_high_temperature;        // C
+  uint16_t             hp_low_temperature;         // C
+  uint16_t             hp_cell_voltage[kNumCells]; // mV
```

Comment:
should we use array<uint16_t, kNumCells> hp_cell_voltage for consistency reasons?

Reply:
Made the change, assigned a value to kFIFO, few other fixes in the new commits and the code compiles on my part now.